src/components/RemoveAnItem.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {addOrRemove, REMOVE_ITEM, REMOVE_ALL} from '../actions/action';
+
+const RemoveAnItem = React.createClass({
+	displayButtons(){
+		return this.props.Store.getState().shoppingCart.map((element, index)=>(<button onClick={this.props.Store.dispatch.bind(this, addOrRemove(REMOVE_ITEM, element.item, element.count))} key={index}>{"Remove " + element.item}</button>));
```

Comment:
way too long - you should separate out functions and variables outside of the return whenever you can.